crates/polars-lazy/src/scan/file_list_reader.rs

```diff
@@ -145,23 +236,9 @@ pub trait LazyFileListReader: Clone {
         None
     }
 
-    /// Get list of files referenced by this reader.
-    ///
-    /// Returns [None] if path is not a glob pattern.
-    fn iter_paths(&self) -> PolarsResult<Option<PathIterator>> {
-        let paths = self.paths();
-        if paths.is_empty() {
-            let path_str = self.path().to_string_lossy();
-            if path_str.contains('*') || path_str.contains('?') || path_str.contains('[') {
-                polars_glob(&path_str, self.cloud_options()).map(Some)
-            } else {
-                Ok(None)
-            }
-        } else {
-            polars_ensure!(self.path().to_string_lossy() == "", InvalidOperation: "expected only a single path argument");
-            // Lint is incorrect as we need static lifetime.
-            #[allow(clippy::unnecessary_to_owned)]
-            Ok(Some(Box::new(paths.to_vec().into_iter().map(Ok))))
-        }
+    /// Returns a list of paths after resolving globs and directories, as well as
+    /// the hive start index.
```

Comment:
What does a `hive start index` mean?

Reply:
refers to where we start to parse hive partitions in the path - added comments here and above the function